### ISSUE TITLE
warning for missing require when using addSingletonGetter

### ISSUE THREAD

**Issue by pstjvn** [google/closure-compiler#1719]
I have this code:

```js
// app.js
goog.module('app');

var ColorWheel = goog.require('pstj.graphics.ColorWheel');
var cw = ColorWheel.getInstance();

// colorwheel.js
goog.module('pstj.graphics.ColorWheel');

var ColorWheel = class ColorWheel {
  constructor() {/** impl */
};
goog.addSingletonGetter(ColorWheel);
exports = ColorWheel;
```

and I get this warning:

```
missing require: 'module$exports$pstj$graphics$ColorWheel.getInstance'
var cw = ColorWheel.getInstance();
         ^
```

Why is this a missing require? I think this might be a bug.

**Comment by MatrixFrog** [comment]
I just saw a similar case recently and @blickly and I were discussing a plan to fix this. Eventually we'll move the checkRequires pass before the goog.module rewriting pass, and that should make this go away. For now, you can make that pass locally in your own copy of the compiler, or turn off the lintChecks group. I would recommend the latter both because it's simpler and because you can get the same checks from the standalone linter, without these false positives.

**Comment by MatrixFrog** [comment]
It looks like this might be fixed already.

https://closure-compiler-debugger.appspot.com/#input0%3Dgoog.module('pstj.graphics.ColorWheel')%253B%250A%250Avar%2520ColorWheel%2520%253D%2520class%2520ColorWheel%2520%257B%250A%2520%2520constructor()%2520%257B%252F**%2520impl%2520*%252F%257D%250A%257D%253B%250Agoog.addSingletonGetter(ColorWheel)%253B%250Aexports%2520%253D%2520ColorWheel%253B%26input1%3Dgoog.module('app')%253B%250A%250Avar%2520ColorWheel%2520%253D%2520goog.require('pstj.graphics.ColorWheel')%253B%250Avar%2520cw%2520%253D%2520ColorWheel.getInstance()%253B%26conformanceConfig%26externs%3Dvar%2520goog%253B%250Agoog.addSingletonGetter%253B%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_REQUIRES%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CLOSURE_PASS%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by Dominator008** [comment]
Looks like this has been fixed. Closing.

**Comment by hochhaus** [comment]
I think the strictMissingRequires check in the v20160713 compiler reintroduces this issue.

**Comment by MatrixFrog** [comment]
Okay I'll try to take another look soon.

**Comment by Dominator008** [comment]
@pstjvn @hochhaus I think this has been fixed (again) ?

**Comment by hochhaus** [comment]
Yes, I have not seen these issues recently.

**Comment by Dominator008** [comment]
@hochhaus Great, thanks for verifying.

**Comment by dashersw** [comment]
I am constantly seeing this issue whenever i try to do `getInstance`.
